src/components/entropy/index.js

```diff
@@ -209,9 +212,24 @@ class Entropy extends React.Component {
     }
     this.setState({chart});
   }
+  visibilityOnScreenChange(entries) {
+    if (entries.length!==1) {
+      return console.error(`Unexpected IntersectionObserver callback entries of length`, entries.length);
+    }
+    const onScreen = entries[0].isIntersecting;
+    if (onScreen===this.props.onScreen) return; // can happen when component initially rendered
+    // if gone off screen or come back on screen with the bars still valid then it's a simple toggle action
+    if (!onScreen || this.props.bars) {
+      return this.props.dispatch({type: ENTROPY_ONSCREEN_CHANGE, onScreen})
+    }
+    // else if back on screen and the bars are invalid then we want to regenerate them
+    this.requestRecalculationOfBars(onScreen);
+  }
```

Comment:
_non-blocking_

Ditto, this feels like an anti-pattern. I would expect the one `ENTROPY_ONSCREEN_CHANGE` action and it would run the entropy re-calculation when appropriate.